pkg/ovirt/install.sh

```diff
@@ -0,0 +1,103 @@
+#      update cockpit-ovirt runtime configuration (to assemble oVirt REST API URL)
+#
+# How to run:
+#      either manually after rpm installations as root:
+#         # cd [COCKPIT_OVIRT_INSTALL_DIR] && ./install.sh https://[ENGINE_HOST]/ovirt-engine/
+#         # "COCKPIT_OVIRT_INSTALL_DIR" usually refers to /usr/share/cockpit/ovirt
+#
+#      or
+#         login into cockpit as the 'root' user
+#         access the 'ovirt' plugin
+#         installation dialog shows up to handle the install.sh script execution from UI
+#
+
+ENGINE_URL=$1
+COCKPIT_DIR=${2:-/usr/share/cockpit}  # TODO: get it dynamically via 'rpm -q cockpit-shell --fileprovide' or 'dpkg -L'
+ETC_COCKPIT_DIR='/etc/cockpit'
+VIRSH_CONNECTION_URI=$3 # optional
+VIRSH_CONNECTION_NAME='remote' # used if VIRSH_CONNECTION_URI is set
+
+EXIT_PARAMS=1 # wrong command parameters
+EXIT_NO_ACCESS_MACHINES_OVIRT_CONFIG=3 # can't write to /etc/cockpit/machines-ovirt.config, try as root
+EXIT_NO_ACCESS_OVIRT_OVERRIDE=4 # can't write to cockpit/ovirt/override.json
+
+function usage() {
+  echo Usage: $0 '[ENGINE_URL] [[COCKPIT_INSTLLATION_DIR]] [[VIRSH_CONNECTION_URI]]'
+  echo Example: $0 https://engine.mydomain.com/ovirt-engine/
+  echo Example: $0 https://engine.mydomain.com/ovirt-engine/ /usr/share/cockpit
+  echo Example: $0 https://engine.mydomain.com/ovirt-engine/ /usr/share/cockpit qemu:///system
+}
+
+function checkParams() {
+  if [ x${ENGINE_URL} = x ] ; then
+    usage
+    exit ${EXIT_PARAMS}
+  else
+    echo Registering for ENGINE_URL: ${ENGINE_URL}
+  fi
+}
+
+function generateProviderConfig() {
+  CONFIG_FILE=${ETC_COCKPIT_DIR}/machines-ovirt.config
+
+  # TODO: decrease polling interval if AuditLog-based updates proof to be working
+  echo "{ \
+      \"debug\": false, \
+      \"ovirt_polling_interval\": 60000, \
+      \"cockpitPort\": 9090, \
+      \"OVIRT_BASE_URL\": \"$ENGINE_URL\"," \
+        > ${CONFIG_FILE} || exit ${EXIT_NO_ACCESS_MACHINES_OVIRT_CONFIG}
+
+  if [ x${VIRSH_CONNECTION_URI} = x ] ; then
+    echo " \
+        \"Virsh\": { \
+            \"connections\": { \
+                \"system\": { \
+                    \"params\": [\"-c\", \"qemu:///system\"] \
+                }, \
+                \"session\": { \
+                    \"params\": [\"-c\", \"qemu:///session\"] \
+                } \
+            } \
+        }" >> ${CONFIG_FILE}
+  else
+    echo " \
+        \"Virsh\": { \
+            \"connections\": { \
+                \"${VIRSH_CONNECTION_NAME}\": { \
+                    \"params\": [\"-c\", \"${VIRSH_CONNECTION_URI}\"] \
+                } \
+            } \
+        }" >> ${CONFIG_FILE}
+  fi
+
+  echo "}" >> ${CONFIG_FILE}
+
+  echo OK: ${CONFIG_FILE} generated
+}
+
+function updateOvirtManifest() {
+  OVIRT_OVERRIDE=${COCKPIT_DIR}/ovirt/override.json
+
+  echo "{ \
+      \"content-security-policy\": \"default-src 'self' data: blob:;connect-src 'self' ws: wss: $ENGINE_URL\" \
```

Comment:
These HTTP accesses should happen via cockpit-bridge and HTTP channels http://cockpit-project.org/guide/latest/cockpit-http.html. Is there a reason for it to happen directly from the web browser?

Reply:
There's no requirement for that. Such a proxy will solve a lot of issues with CORS and even the post-installation step can be much simpler. Thanks for pointing me to it.